### DISCUSSION TITLE
Main Exception Message: unknown error: no chrome binary at /Users/s0k04qc/Documents/educative-courses/educative.io_scraper/Chrome-bin/mac/Chromium.app/Contents/MacOS/Google Chrome

### DISCUSSION THREAD

**Discussion by sarveshkhanna** [anilabhadatta/educative.io_scraper#27] [Q&A] [answered]
getting error mentioning cannot find chrome binary for chrome

**Comment by anilabhadatta** [comment]
did u download the chrome binary from releases? @sarveshkhanna

**Reply by sarveshkhanna** [reply]
yes downloaded from :
https://google-chrome.en.uptodown.com/mac/download/4499193

**Reply by anilabhadatta** [reply]
Please download the binary from here: https://github.com/anilabhadatta/educative.io_scraper/releases/download/Educative-Scraper-v6.8/Chrome-bin-osx.zip

If the chrome version is automatically updated then download the chromedriver from here and replace the chromedriver from Chrome-driver/mac. Download the correction version of chromedriver that supports your OS and Chrome versio.

**Reply by sarveshkhanna** [reply]
which chrome version is this compatible with? 99 is fine?

**Reply by sarveshkhanna** [reply]
hi @anilabhadatta  after adding the above driver able to login but the scrapper fails on the first page:
0 https://www.educative.io/courses/grokking-adv-system-design-intvw/YQ49jQx9Ry2 
list index out of range
any way out

**Reply by anilabhadatta** [reply]
@sarveshkhanna please send screenshot of terminal

**Reply by sarveshkhanna** [reply]
![Screenshot 2023-01-10 at 10 34 02 AM](https://user-images.githubusercontent.com/9054658/211466256-809b5c3c-c3ff-4681-b86d-7677fe7ab26d.png)

**Reply by sarveshkhanna** [reply]
@anilabhadatta  for your reference

**Reply by anilabhadatta** [reply]
@sarveshkhanna working at my end, please clone the latest repository and test it again.

**Reply by anilabhadatta** [reply]
Scraper Started, Log file can be found in Save directory

Driver Loaded

                            [Selected config: 0] Starting Scraping: 0, https://www.educative.io/courses/grokking-adv-system-design-intvw/YQ49jQx9Ry2

Load Webpage Function
Checking Login Function
Checking for captcha Function...
Create Course Folder Function
Getting File Name
> This is a regular page
Get File name standard
Inside Course Folder
Checking Login Function
Checking for captcha Function...
Scrolling Page
Getting File Name
> This is a regular page
Get File name standard
Checking page
Removing Nav tags from page
Node deleted div[class*='ed-grid'] > nav
Node deleted div[class*='ed-grid'] > nav
Show Hints Function
No hints found
Inside Mark Down Quiz function
No mark down quiz_container found
Show Codebox Answers Function
No Codebox answers found
Finding Slides Function
No Slides Found
Inside take_quiz_screenshot function
Quiz not found
Adding Name Tag in Next Back Button
Fixing SVG Tags inside Object Tags
Get HTML Page Content Using Single File Function
make_code_selectable function
make_code_selectable function executed
Creating HTML File
HTML File Created
Take Screenshot Function
Screenshot taken and HTML File generated
HTML Page content taken.
Inside Widget Container Function
No widget container found
Code Container Download Type Function
No Code Container Downloadable Type found
Code Container Clipboard Type Function
No code containers found
Remove Mark completed
Next Page Function
Going Next Page


Manually tested the Single File HTML generator and Screenshot HTML generator function.

**Reply by anilabhadatta** [reply]
@sarveshkhanna Can you check the readme in your cloned directory and tell me the Repo version?
Also sometimes due to slow internet, the scraper might cause some issues. Also, could you try running the scraper in Non-Headless mode to see if the account is logged in?

**Comment by anilabhadatta** [comment, marked as answer]
@sarveshkhanna latest version should fix your issue, do check it
